File: tui xach lvt/backend/src/main/java/com/tuixach/lvt/entity/Review.java
package com.tuixach.lvt.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "reviews", uniqueConstraints = {
    @UniqueConstraint(name = "uk_review_user_product", columnNames = { "user_id", "product_id" })
}, indexes = {
    @Index(name = "idx_review_product_created", columnList = "product_id,created_at")
})
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Review {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "product_id", nullable = false)
    private Product product;

    @Column(nullable = false)
    private int rating; // 1-5

    @Column(columnDefinition = "TEXT")
    private String comment;

    @Column(nullable = false)
    private boolean verifiedPurchase = false;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "order_item_id")
    private OrderItem orderItem;

    @Column(nullable = false)
    private boolean active = true;

    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }

    @OneToMany(mappedBy = "review", cascade = CascadeType.ALL, orphanRemoval = true)
    private java.util.List<ReviewImage> images;
}

File: tui xach lvt/frontend/src/app/core/models/review.ts
export interface Review {
  id: number;
  userId: number;
  userName: string;
  productId: number;
  productName: string;
  orderItemId?: number;
  verifiedPurchase: boolean;
  rating: number;
  comment: string;
  images?: string[];
  createdAt: string;
  active?: boolean;
}

export interface ReviewSummary {
  averageRating: number;
  totalCount: number;
  distribution: Record<string, number>;
}

export interface ReviewEligibility {
  eligible: boolean;
  orderItemId?: number;
  reason?: 'NOT_PURCHASED' | 'ALREADY_REVIEWED' | 'NOT_DELIVERED' | string;
}
